db-service/test/cqn4sql/calculated-elements.test.js

```diff
@@ -701,19 +701,23 @@ describe('Unfolding calculated elements in select list', () => {
     expect(query).to.deep.equal(expected)
   })
 
-  it('exists cannot leverage calculated elements', () => {
+  it('exists cannot leverage calculated elements which ends in string', () => {
     // at the leaf of a where exists path, there must be an association
-    // calc elements can't end in an association, hence this does not work, yet.
     expect(() => cqn4sql(CQL`SELECT from booksCalc.Books { ID } where exists youngAuthorName`, model)).to.throw(
-      `Calculated elements cannot be used in “exists” predicates in: “exists youngAuthorName”`,
+      `Expecting path “youngAuthorName” following “EXISTS” predicate to end with association/composition, found “cds.String”`,
+    )
+  })
+  it('exists cannot leverage calculated elements which is an expression', () => {
+    // at the leaf of a where exists path, there must be an association
+    expect(() => cqn4sql(CQL`SELECT from booksCalc.Books { ID } where exists authorFullName`, model)).to.throw(
+      `Expecting path “authorFullName” following “EXISTS” predicate to end with association/composition, found “expression”`,
```

Comment:
Should we just omit the `, found “expression”` suffix, if the type cannot be determined?

Reply:
I can do it, why not use the nice prosa :)